Simplenote/NoteContentHelper.swift

```diff
@@ -0,0 +1,89 @@
+        let titleRange = self.titleRange(in: content)
+        return (
+            title: titleRange,
+            body: bodyRange(in: content, titleRange: titleRange)
+        )
+    }
+
+    private static func titleRange(in content: String?) -> NSRange {
+        guard let content = content else {
+            return NSRange(location: NSNotFound, length: 0)
+        }
+        let fullRange = content.nsString.fullRange
+
+        // Look for the first character ignoring whitespaces and newlines
+        let firstCharacterRange = content.nsString.rangeOfCharacter(from: CharacterSet.whitespacesAndNewlines.inverted,
+                                                                    options: [],
+                                                                    range: fullRange)
+        guard firstCharacterRange.location != NSNotFound else {
+            return firstCharacterRange
+        }
+
+        // Look for the next newline
+        let newlineSearchRange = NSRange(location: firstCharacterRange.location,
+                                         length: fullRange.length - firstCharacterRange.location)
+        let newlineRange = content.nsString.rangeOfCharacter(from: .newlines,
+                                                             options: [],
+                                                             range: newlineSearchRange)
+        guard newlineRange.location != NSNotFound else {
+            return newlineSearchRange
+        }
+
+        return rangeByTrimmingTrailingWhitespacesAndNewlines(in: content,
+                                                             firstCharacterLocation: firstCharacterRange.location,
+                                                             endRangeLocation: newlineRange.location)
+    }
+
+    private static func bodyRange(in content: String?, titleRange: NSRange) -> NSRange {
+        guard titleRange.location != NSNotFound, let content = content else {
+            return titleRange
+        }
+
+        let fullRange = content.nsString.fullRange
+
+        // Look for the first character after title
+        let untrimmedBodyRange = NSRange(location: NSMaxRange(titleRange),
+                                         length: fullRange.length - NSMaxRange(titleRange))
+        let firstCharacterRange = content.nsString.rangeOfCharacter(from: CharacterSet.whitespacesAndNewlines.inverted,
+                                                                    options: [],
+                                                                    range: untrimmedBodyRange)
+        guard firstCharacterRange.location != NSNotFound else {
+            return firstCharacterRange
+        }
+
+        return rangeByTrimmingTrailingWhitespacesAndNewlines(in: content,
+                                                             firstCharacterLocation: firstCharacterRange.location,
+                                                             endRangeLocation: content.fullRange.length)
+    }
```

Comment:
Second note: (IF possible), it'd be neat to replace `NSString+Condensing: generatePreviewStrings` with your new API, so that a single component builds the Title / Body

Reply:
1. Managed to merge them :-)
2. My reason: I wanted to keep all the logic regarding what we consider "title" and "body" in one place
3. Agree with that!